contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -387,31 +389,38 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
 
     // The Functions Router will perform the callback to the client contract
     IFunctionsRouter router = IFunctionsRouter(address(s_router));
-    (uint8 result, uint96 callbackCostJuels) = router.fulfill(
-      requestId,
-      response,
-      err,
-      uint96(juelsPerGas),
-      costWithoutFulfillment,
-      msg.sender
-    );
-
-    // Reimburse the transmitter for the fulfillment gas cost
-    s_withdrawableTokens[msg.sender] = gasOverheadJuels + callbackCostJuels;
-    // Put donFee into the pool of fees, to be split later
-    // Saves on storage writes that would otherwise be charged to the user
-    s_feePool += commitment.donFee;
-
-    emit BillingEnd(
-      requestId,
-      commitment.subscriptionId,
-      commitment.donFee,
-      gasOverheadJuels + callbackCostJuels,
-      gasOverheadJuels + callbackCostJuels + commitment.donFee + commitment.adminFee,
-      FulfillResult(result)
-    );
-
-    return FulfillResult(result);
+    try router.fulfill(requestId, response, err, uint96(juelsPerGas), costWithoutFulfillment, msg.sender) returns (
+      uint8 result,
+      uint96 callbackCostJuels
+    ) {
+      FulfillResult resultCode = FulfillResult(result);
+      if (resultCode == FulfillResult.USER_SUCCESS || resultCode == FulfillResult.USER_ERROR) {
+        // Reimburse the transmitter for the fulfillment gas cost
+        s_withdrawableTokens[msg.sender] = gasOverheadJuels + callbackCostJuels;
+        // Put donFee into the pool of fees, to be split later
+        // Saves on storage writes that would otherwise be charged to the user
+        s_feePool += commitment.donFee;
+
+        emit BillingEnd(
+          requestId,
+          commitment.subscriptionId,
+          commitment.donFee,
+          gasOverheadJuels + callbackCostJuels,
+          gasOverheadJuels + callbackCostJuels + commitment.donFee + commitment.adminFee,
+          FulfillResult(result)
+        );
+      }
+
+      return resultCode;
+    } catch Error(string memory reason) {
```

Comment:
This is roughly equivalent to what we had in v0, right?

Reply:
Yes, similar